src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int size;
+
     @Override
     public void add(T value) {
+        if (firstNode == null) {
+            Node<T> newNode = new Node<>(null, value, null);
+            firstNode = newNode;
+            lastNode = newNode;
+        } else {
+            lastNode.next = new Node<>(lastNode, value, null);
+            lastNode = lastNode.next;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index out of bounds");
+        }
+        if (index == size) {
+            add(value);
+            return;
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, firstNode);
+            if (firstNode != null) {
+                firstNode.prev = newNode;
+            }
+            firstNode = newNode;
+            if (lastNode == null) {
+                lastNode = newNode;
+            }
```

Comment:
In the `add(T value, int index)` method, the logic for adding at the beginning of the list is correct but could be simplified by directly setting `firstNode.prev` to `newNode` without the null check, as `firstNode` is guaranteed to be non-null when `index == 0`.